gno.land/pkg/gnoweb/frontend/js/controller-action-function.ts

```diff
@@ -211,7 +211,12 @@ export class ActionFunctionController extends BaseController {
 	// Fetch the qeval result from the remote
 	private async _fetchQEval(remote: string, data: string): Promise<string> {
 		try {
-			const url = `http://${remote}/abci_query?path=vm%2fqeval&data=${btoa(data)}`;
+			// If remote already has a protocol, use it as-is; otherwise prepend http://
+			const baseUrl =
+				remote.startsWith("http://") || remote.startsWith("https://")
```

Comment:
It's valid to start gno web with `-remote tcp://127.0.0.1:26657` . The `tcp` form works in gnokey. Should we also allow `tcp`?